packages/fxa-settings/src/components/PageDeleteAccount/index.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { FormEvent, useCallback, useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { RouteComponentProps, useNavigate } from '@reach/router';
+import { useAccountDestroyer } from '../../lib/auth';
+import { sessionToken } from '../../lib/cache';
+import { useAlertBar } from '../../lib/hooks';
+import { useAccount } from '../../models';
+import InputPassword from '../InputPassword';
+import FlowContainer from '../FlowContainer';
+import VerifiedSessionGuard from '../VerifiedSessionGuard';
+import AlertBar from '../AlertBar';
+import { HomePath } from '../../constants';
+import { logViewEvent, usePageViewEvent } from '../../lib/metrics';
+import { Checkbox } from '../Checkbox';
+import { cloneDeep } from '@apollo/client/utilities';
+
+type FormData = {
+  password: string;
+};
+
+export const PageDeleteAccount = (_: RouteComponentProps) => {
+  usePageViewEvent('settings.delete-account');
+  const { handleSubmit, register, formState, setValue } = useForm<FormData>({
+    mode: 'all',
+    defaultValues: {
+      password: '',
+    },
+  });
+  // TODO: the `.errors.password` clause shouldn't be necessary, but `isValid` isn't updating
+  // properly. I think this is a bug in react-hook-form.
+  const disabled =
+    !formState.isDirty || !formState.isValid || !!formState.errors.password;
+  const [errorText, setErrorText] = useState<string>();
+  const [confirmed, setConfirmed] = useState<boolean>(false);
+  const [hideContinueButton, setHideContinueButton] = useState<boolean>(true);
+  const [accountDeleted, setAccountDeleted] = useState<boolean>(false);
+  const [subtitleText, setSubtitleText] = useState<string>('Step 1 of 2');
+  const navigate = useNavigate();
+  const alertBar = useAlertBar();
+  const goBack = useCallback(() => window.history.back(), []);
+
+  const { primaryEmail } = useAccount();
+
+  const advanceStep = () => {
+    setSubtitleText('Step 2 of 2');
+    setConfirmed(true);
+
+    logViewEvent('flow.settings.account-delete', 'terms-checked.success');
+  };
+
+  const deleteAccount = useAccountDestroyer({
+    onSuccess: () => {
+      setAccountDeleted(true);
+
+      logViewEvent('flow.settings.account-delete', 'confirm-password.success');
+    },
+    onError: (error) => {
+      if (error.errno === 103) {
+        setErrorText(error.message);
+        setValue('password', '');
+      } else {
+        alertBar.setType('error');
+        alertBar.setContent(error.message);
+        alertBar.show();
+        logViewEvent('flow.settings.account-delete', 'confirm-password.fail');
+      }
+    },
+  });
+
+  const onFormSubmit = ({ password }: FormData) => {
+    deleteAccount.execute(primaryEmail.email, password, sessionToken()!);
+  };
+
+  return (
+    <>
+      {accountDeleted ? (
+        <FlowContainer title="Delete Account">
+          Your account has been deleted
+        </FlowContainer>
+      ) : (
+        <FlowContainer title="Delete Account" subtitle={subtitleText}>
```

Comment:
Should distinct flows go into their own file or is it up to the developer to make that call? Having distinct files might make the code more readable, but probably introduces other problems.

Reply:
We don't have a decision on this, I followed `PageRecoveryKeyAdd` which has them all in the same document. Would be nice to have a short discussion on it

Reply:
Heh yea that would probably be good. 

Some quick thoughts of mine:

* Having all the views in one file make sense if its a wizard type page, except maybe if there are 10 sections?
* Maybe introduce a wizard page component
* Code is more readable in separate files but I don't believe our folder structure would scale well